File: app/src/main/java/es/pau/calculadoraoposiciones/Preferences.java
package es.pau.calculadoraoposiciones;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Project: Calculadoradeprobabilidaddeoposiciones
 * Created on 22/06/16.
 */
public class Preferences {

    private static String SHARED_PREFERENCES = "shared preferences";

    public static final String TOTAL_SAVED = "total topics saved";
    public static final String TAKEN_SAVED = "taken topics saved";
    public static final String STUDIED_SAVED = "studied topics saved";
    public static final String RESULT_SAVED = "percentage resulted saved";

    private static SharedPreferences sharedPreferences;

    public static void register(Context context) {
        sharedPreferences = context.getSharedPreferences(SHARED_PREFERENCES, Context.MODE_PRIVATE);
    }

    public static boolean loadBoolean(String key) {
        return sharedPreferences.getBoolean(key, false);
    }

    public static void saveBoolean(String key, Boolean value) {
        sharedPreferences.edit().putBoolean(key, value).apply();
    }

    public static int loadInt(String key) {
        return sharedPreferences.getInt(key, 0);
    }

    public static void saveInt(String key, int value) {
        sharedPreferences.edit().putInt(key, value).apply();
    }

    public static String loadString(String key) {
        return sharedPreferences.getString(key, "");
    }

    public static void saveString(String key, String value) {
        sharedPreferences.edit().putString(key, value).apply();
    }

    public static void clear() {
        sharedPreferences.edit().clear().apply();
    }

    public static void delete(String key) {
        sharedPreferences.edit().remove(key).apply();
    }

    public static boolean hasPreference(String key) {
        return sharedPreferences.contains(key);
    }
}

File: app/src/main/java/es/pau/calculadoraoposiciones/utils/Preferences.kt
package es.pau.calculadoraoposiciones.utils

import android.content.Context
import android.content.SharedPreferences

/**
 * Created on 13/06/18.
 *
 * Based on the very clever solution of Ben Deitch from http://blog.teamtreehouse.com/making-sharedpreferences-easy-with-kotlin
 */
class Preferences (context: Context) {
    private val PREFS_FILENAME = "es.pau.calculadoraoposiciones.prefs"
    private val prefs: SharedPreferences = context.getSharedPreferences(PREFS_FILENAME, 0);

    fun remove(key: String) = prefs.edit().remove(key).apply()

    fun contains(key: String) = prefs.contains(key)

    // App Preferences

    val TOTAL_SAVED = "total topics saved"
    var totalPref: Int
        get() = prefs.getInt(TOTAL_SAVED, 0)
        set(value) = prefs.edit().putInt(TOTAL_SAVED, value).apply()

    val TAKEN_SAVED = "taken topics saved"
    var takenPref: Int
        get() = prefs.getInt(TAKEN_SAVED, 0)
        set(value) = prefs.edit().putInt(TAKEN_SAVED, value).apply()

    val STUDIED_SAVED = "studied topics saved"
    var studiedPref: Int
        get() = prefs.getInt(STUDIED_SAVED, 0)
        set(value) = prefs.edit().putInt(STUDIED_SAVED, value).apply()

    val RESULT_SAVED = "percentage resulted saved"
    var resultPref: String
        get() = prefs.getString(RESULT_SAVED, "")
        set(value) = prefs.edit().putString(RESULT_SAVED, value).apply()
}